src/NUnitTestAdapter/NUnitEngine/NUnitTestEvent.cs

```diff
@@ -199,8 +189,25 @@ public NUnitTestEventTestCase(XmlNode node) : base(node)
 
         public string ReasonMessage { get; }
 
-        public bool HasReason => string.IsNullOrEmpty(ReasonMessage);
+        public bool HasReason => !string.IsNullOrEmpty(ReasonMessage);
         public bool HasFailure => Failure != null;
+
+        /// <summary>
+        /// Find stacktrace in assertion nodes if not defined (seems .netcore2.0 doesn't provide stack-trace for Assert.Fail("abc"))
+        /// </summary>
+        public string StackTrace
+        {
+            get
+            {
+                string stackTrace = string.Empty;
+                foreach (XmlNode assertionStacktraceNode in Node.SelectNodes("assertions/assertion/stack-trace"))
+                {
+                    stackTrace += assertionStacktraceNode.InnerText;
+                }
```

Comment:
If there are multiple assertions, we should show separate stack traces rather than concatenating them as though there was one call inside all the other calls.

Reply:
I am unsure how the engine creates these nodes, and what actually they represent.  Also need to figure out if we have that option in the Test Explorer interface.

Reply:
I believe this is tied to the multiple assertions feature in the framework:

```cs
Assert.Multiple(() =>
{
    Assert.Fail(); // Here's one stack trace

    Assert.That(1, Is.EqualTo(2)); // Here's another stack trace.
});
```

Reply:
We're currently returning this a single test resultset, and there we only have a single stack trace property.  Believe we need to look more into how to represent this on the Test Explorer side.